pkg/controller/controller.go

```diff
@@ -427,3 +434,52 @@ func (c *Controller) collectConfigMap() error {
 
 	return nil
 }
+func cloneFunction(in interface{}) (*spec.Function, error) {
+	var scheme = runtime.NewScheme()
+	clone, err := scheme.DeepCopy(in)
+	if err != nil {
+		return nil, err
+	}
+
+	out, ok := clone.(*spec.Function)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type: %T", clone)
+	}
+
+	return out, nil
+}
+
+func (c *Controller) updateFunctionStatus(funcObj *spec.Function) error {
+
+	funcObjCopy, err := cloneFunction(funcObj)
```

Comment:
It is not clear to me why you need to copy the function object. And if it is needed anyway, wouldn't this be enough?
```
	funcObjCopy := spec.Function{}
	funcObjCopy = *funcObj
```

Reply:
#5 https://github.com/kubernetes/community/blob/master/contributors/devel/controllers.md

Reply:
I see